credentialdigger/cli/get_discoveries.py

```diff
@@ -0,0 +1,276 @@
+        discoveries_df = pd.DataFrame(discoveries)
+        # Remove the repo_url column since it has been passed as an argument
+        del discoveries_df['repo_url']
+        # Remove timestamp and rule_id columns because they are not
+        # quite relevant
+        del discoveries_df['timestamp']
+        del discoveries_df['rule_id']
+
+        # Convert `int` columns to `str` to be eventually rendered.
+        discoveries_df['id'] = discoveries_df['id'].astype(str)
+        discoveries_df['line_number'] = discoveries_df['line_number'].astype(
+            str)
+
+        # Convert to list and insert column names
+        discoveries_list = discoveries_df.values.tolist()
+        columns = ['id', 'file_name', 'commit_id', 'line_number',
+                   'snippet', 'state']
+
+        table = Table(
+            title=f'Discoveries found in "{repo_url}"',
+            pad_edge=False, show_lines=True)
+
+        for c in columns:
+            table.add_column(c)
+        for row in discoveries_list:
+            table.add_row(*row)
+
+        console.print(table)
+
+
+def discoveries_to_csv(discoveries):
+    """ Generate CSV from list of discoveries
+
+    Parameters
+    ----------
+    discoveries: list
+        List of discoveries from which to generate the CSV
+
+    Returns
+    -------
+    str
+        A string containing CSV obtained from the original list of
+        discoveries.
+    """
+    try:
+        stringIO = io.StringIO()
+        csv_writer = csv.DictWriter(stringIO, discoveries[0].keys())
+        csv_writer.writeheader()
+        csv_writer.writerows(discoveries)
+        csv_data = stringIO.getvalue()
+    except IndexError as error:
+        logger.error(error)
+    except Exception as exception:
+        logger.exception(exception)
+
+    return csv_data
+
+
+def export_csv(discoveries, client, save=False):
+    """ Export discoveries as a CSV file
+
+    Parameters
+    ----------
+    discoveries: list
+        List of discoveries from which to generate the CSV
+    client: `credentialdigger.Client`
+        Instance of the client from which we retrieve rules
+    save: bool
+        If True, we do not ask the user to enter a file path for the CSV
+        to be exported
+    """
+    # Check if --save is specified
+    if save is False:
+        path = ''
+        # Read the export path from the console's input
+        while path == '':
+            path = console.input('Path to export CSV:')
+    else:
+        # if --save argument is set, we use it as an export path
+        path = save
+
+    try:
+        csv_file = open(path, newline='', mode='w')
+    except IOError as e:
+        console.print(f'[red]{e}[/]')
+    else:
+        with csv_file:
+            with console.status('[bold]Exporting the discoveries..'):
+                # Add the category to each discovery
+                assign_categories(client, discoveries)
```

Comment:
I suggest not to assign the category, at the moment. Maybe we'll introduce it later, but since it's not even in the `discoveries` table I think we can avoid it.